common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsManager.kt

```diff
@@ -0,0 +1,101 @@
+package com.revenuecat.purchases.common.diagnostics
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.Dispatcher
+import com.revenuecat.purchases.common.verboseLog
+import java.io.IOException
+
+class DiagnosticsManager(
+    private val diagnosticsFileHelper: DiagnosticsFileHelper,
+    private val diagnosticsAnonymizer: DiagnosticsAnonymizer,
+    private val backend: Backend,
+    private val diagnosticsDispatcher: Dispatcher,
+    private val sharedPreferences: SharedPreferences
+) {
+    companion object {
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        const val CONSECUTIVE_FAILURES_COUNT_KEY = "consecutive_failures_count"
+        @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+        const val MAX_NUMBER_RETRIES = 3
+
+        fun initializeSharedPreferences(context: Context): SharedPreferences =
+            context.getSharedPreferences(
+                "com_revenuecat_purchases_${context.packageName}_preferences_diagnostics",
+                Context.MODE_PRIVATE
+            )
+    }
+
+    fun syncDiagnosticsFileIfNeeded() {
+        enqueue {
+            try {
+                val diagnosticsList = diagnosticsFileHelper.readDiagnosticsFile()
+                val diagnosticsCount = diagnosticsList.size
+                if (diagnosticsCount == 0) {
+                    verboseLog("No diagnostics to sync.")
+                    return@enqueue
+                }
+                backend.postDiagnostics(
+                    diagnosticsList = diagnosticsList,
+                    onSuccessHandler = {
+                        verboseLog("Synced diagnostics file successfully.")
+                        clearConsecutiveNumberOfErrors()
+                        diagnosticsFileHelper.cleanSentDiagnostics(diagnosticsCount)
+                    },
+                    onErrorHandler = { error, shouldRetry ->
+                        verboseLog("Error syncing diagnostics file: $error. Should retry: $shouldRetry")
+                        if (shouldRetry) {
+                            val currentRetryCount = increaseConsecutiveNumberOfErrors()
+                            if (currentRetryCount >= MAX_NUMBER_RETRIES) {
+                                resetDiagnosticsStatus()
+                            }
+                        } else {
+                            resetDiagnosticsStatus()
+                        }
```

Comment:
I wonder if this would be simpler like:
```suggestion
                        if (shouldRetry || increaseConsecutiveNumberOfErrors() >= MAX_NUMBER_RETRIES) {
                            resetDiagnosticsStatus()
                        }
```

Reply:
In the end, I added the logs Andy suggested below, so didn't do this change